lib/shopify-cli/process_supervision.rb

```diff
@@ -104,24 +113,30 @@ def running?(identifier)
       end
     end
 
-    def initialize(identifier, pid:, time: Time.now.strftime('%s')) # :nodoc:
+    def initialize(identifier, pid: nil, time: Time.now.strftime('%s')) # :nodoc:
       @identifier = identifier
       @pid = pid
       @time = time
+
+      FileUtils.mkdir_p(RUN_DIR)
       @pid_path = File.join(RUN_DIR, "#{identifier}.pid")
       @log_path = File.join(RUN_DIR, "#{identifier}.log")
     end
 
     ##
     # will attempt to shutdown a running process
     #
+    # #### Parameters
+    #
+    # * `ctx` - the context of this command
+    #
     # #### Returns
     #
     # * `stopped` - [true, false]
     #
-    def stop
+    def stop(ctx)
```

Comment:
If you are just passing in a default context above, maybe this should just have a default value

Reply:
That's true, all of these contexts are only used to check if we're running on Windows (no messages or anything), so we don't have to pass in any objects - let's just go with `Context.new.windows?` as you suggested.